common/http/platform/beast/http.cpp

```diff
@@ -199,9 +206,23 @@ void Client::ResolveHandler(
 
 	resolver_results_ = results;
 
+	stream_ =
+		make_shared<beast::ssl_stream<beast::tcp_stream>>(GetAsioIoContext(event_loop_), ssl_ctx_);
+
+	http_response_parser_ = make_shared<http::response_parser<http::buffer_body>>();
+
+	// Don't enforce limits. Since we stream everything, limits don't generally apply, and
+	// if they do, they should be handled higher up in the application logic.
+	//
+	// Note: There is a bug in Beast here (tested on 1.74): One is supposed to be able to
+	// pass an uninitialized `optional` to mean unlimited, but they do not check for
+	// `has_value()` in their code, causing their subsequent comparison operation to
+	// misbehave. So pass highest possible value instead.
+	http_response_parser_->body_limit(numeric_limits<uint64_t>::max());
+
 	weak_ptr<Client> weak_client(stream_active_);
 
-	beast::get_lowest_layer(this->stream_)
+	beast::get_lowest_layer(*this->stream_)
```

Comment:
I'd prefer parentheses here.

Reply:
I just removed `this` altogether. It was the only instance which looked like this, all the others use implicit `this`.